test/hw5/test_task2.py

```diff
@@ -0,0 +1,15 @@
+from All_home_works.hw5.task2 import print_result
+
+
+def foo():
+    """DOCS OF FOO"""
+    return "return foo"
+
+
+def test_doc_string():
+
+    foo_with_decorator = print_result(foo)
+    foo_with_decorator()
```

Comment:
Do you need to call it?

Reply:
I made mistake. I appropriated doc and name before the function had been ever called which explains why it worked only after called.

Reply:
> which explains why it worked only after called

It's worth exploring such strange cases before requesting review.